src/foam/nanos/auth/UserAndGroupAuthService.js

```diff
@@ -257,7 +252,7 @@ foam.CLASS({
     {
       name: 'validatePassword',
       javaCode: `
-        if ( SafetyUtil.isEmpty(potentialPassword) || ! (Pattern.compile(PASSWORD_VALIDATE_REGEX)).matcher(potentialPassword).matches() ) {
+        if ( SafetyUtil.isEmpty(potentialPassword) || potentialPassword.length() < 6 ) {
```

Comment:
This method shouldn't be needed at all. If the userDAO has a ValidatingDAO around it, you shouldn't need to do any of this because the user with the updated password will be put into the dao and rejected.


...after looking closer at this service: It does password hashing before putting into the DAO so the DAO will allow it because it will likely hash to something that is valid. I don't think this service should be responsible for that. We already have a UserPasswordHashingDAO which should be doing this. If that decorator comes after the ValidatingDAO then I think you're good to go.

Reply:
I have no complaints , I opened a ticket to clean passwords. CPF-2259